python_modules/dagster/dagster/_core/definitions/asset_daemon_context.py

```diff
@@ -172,6 +173,21 @@ def target_asset_keys_and_parents(self) -> AbstractSet[AssetKey]:
     def respect_materialization_data_versions(self) -> bool:
         return self._respect_materialization_data_versions
 
+    def prefetch_updated_parents(self) -> None:
+        """Pre-populate the cached values here to avoid situations in which the new latest_storage_id
+        value is calculated a long time after we calculate the set of updated parents for a given
+        asset, as this can cause us to miss materializations.
+        """
+        for asset_key in self.target_asset_keys:
+            self.instance_queryer.asset_partitions_with_newly_updated_parents(
+                latest_storage_id=self.latest_storage_id, child_asset_key=asset_key
+            )
+        self.get_latest_storage_id()
```

Comment:
should the order of these change? if you have to pick it seems like you would want to potentially consider a materialization twice vs. skipping materializations

Reply:
(i.e. seems like you would want to err on the side of the stored latest storage ID being earlier than later)

Reply:
updated!